velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -180,12 +180,26 @@ struct ArrayJoinFunction {
   }
 
   template <typename C>
-  void writeValue(out_type<velox::Varchar>& result, const C& value) {
+  void writeValue(
+      out_type<velox::Varchar>& result,
+      const C& value,
+      const TypePtr& type) {
     // To VARCHAR converter never throws.
+    if constexpr (std::is_same_v<C, int32_t>) {
```

Comment:
nit: cant we just specialisse writeValue so we can get rid of the is_same<C, int32_t> check ?
void writeValue<int32_t>(...) {..}